cep18/src/entry_points.rs

```diff
@@ -0,0 +1,221 @@
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `increase_allowance` entry point.
+pub fn increase_allowance() -> EntryPoint {
+    EntryPoint::new(
+        String::from(INCREASE_ALLOWANCE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(SPENDER, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `decrease_allowance` entry point.
+pub fn decrease_allowance() -> EntryPoint {
+    EntryPoint::new(
+        String::from(DECREASE_ALLOWANCE_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(SPENDER, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `transfer` entry point.
+pub fn transfer() -> EntryPoint {
+    EntryPoint::new(
+        String::from(TRANSFER_ENTRY_POINT_NAME),
+        vec![
+            Parameter::new(RECIPIENT, Key::cl_type()),
+            Parameter::new(AMOUNT, U256::cl_type()),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `balance_of` entry point.
+pub fn balance_of() -> EntryPoint {
+    EntryPoint::new(
+        String::from(BALANCE_OF_ENTRY_POINT_NAME),
+        vec![Parameter::new(ADDRESS, Key::cl_type())],
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `total_supply` entry point.
+pub fn total_supply() -> EntryPoint {
+    EntryPoint::new(
+        String::from(TOTAL_SUPPLY_ENTRY_POINT_NAME),
+        Vec::new(),
+        U256::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `decimals` entry point.
+pub fn decimals() -> EntryPoint {
+    EntryPoint::new(
+        String::from(DECIMALS_ENTRY_POINT_NAME),
+        Vec::new(),
+        u8::cl_type(),
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `burn` entry point.
+pub fn burn() -> EntryPoint {
+    EntryPoint::new(
+        String::from(BURN_ENTRY_POINT_NAME),
+        vec![Parameter::new(AMOUNT, U256::cl_type())],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `mint` entry point.
+pub fn mint() -> EntryPoint {
+    EntryPoint::new(
+        String::from(MINT_ENTRY_POINT_NAME),
+        vec![Parameter::new(AMOUNT, U256::cl_type())],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
+    )
+}
+
+/// Returns the `change_security` entry point.
+pub fn change_security() -> EntryPoint {
+    EntryPoint::new(
+        String::from(CHANGE_SECURITY_ENTRY_POINT_NAME),
+        vec![],
```

Comment:
That is a good question. The only other contract where we use optional entrypoint arguments is cep-78s installation. I'm not sure we currently have an adequate way to indicate optional arguments, since the way we could write it would be the same as `Option` arguments .

For the onlookers sake: `Option<T>` arguments are still required arguments, and missing them would result in an error. On the other hand "optional arguments" allow for the argument to be not provided on execution.

`Option<T>` args still use the default `runtime::get_named_arg` function.
"Optional arguments" use a manually implemented getter that bypasses the automatic reverting in the default implementation that happens when the argument is missing during execution.

Reply:
make sense, we should have standard for this.
cc @darthsiroftardis